qiskit/synthesis/evolution/lie_trotter.py

```diff
@@ -78,6 +78,7 @@ def __init__(
             | None
         ) = None,
         wrap: bool = False,
+        reorder: bool = False,
```

Comment:
We can discuss this, but I think this should be `True` per default as it can significantly optimize the circuit depth 🙂 It will change the output of the current circuits, but as this is a synthesis routine which only promises to output a circuit implementing the Trotter formula, changes like this are fine I think.

Reply:
I disagree. We should only consider performing Pauli term reordering by default for a given Hamiltonian, if we
- understand the potential classical runtime implications
- (strongly) suspect we can gain reductions in circuit depth by inspecting the Hamiltonian
- care about circuit depth at all as specified by some transpilation option
- leveraged Rust for getting the most efficient implementation
The underlying combinatorial problem is NP-complete and while heuristics yield a great runtime improvement they can still be quite slow compared to the rest of the transpilation steps, especially on larger Hamiltonians. Otherweise, I love the PR, thanks @altaris !

I'm mentioning @ajavadia in case he wants to chime in. :-)

Reply:
Runtime cost is a good point 👍🏻 it would be interesting to see the that for large scale Hamiltonians, as in most cases they have an underlying structure. Maybe we could check some from the Hamlib? 🙂

Reply:
Is there a benchmark such as Benchpress HamLib https://github.com/Qiskit/benchpress/tree/main that indicates the CX cost/depth improvement and the time consumption ?

Reply:
Conclusion: we're not enabling this per default for now.